### ISSUE TITLE
Markdown formatting breaks long list in YAML frontmatter for Hugo

### ISSUE THREAD

**Issue by jamesbraza** [prettier/prettier#15187]
**Prettier 3.0.0**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBadAdWBDA5gZyQAIBtDEbSq6m7cgGiPICNW32OGnwxe++u5ACYiRgkHElTpccgF0s6VFhD0QEAA4wAltHzJQ2AE5GIAdwAKxhPpTYANmewBPfWuZHsYANZwYAZWwAWzgAGW0oOGQAMwd8OHdPHz9-DS8I3GQYIwBXBJB4oO0s3Pz8DPs4AEUciHgYuPyAK3wAD38K6tr6pFj7eLUAR264C1MNWwp8VEi4ITnVEGzsbXsMgGEIIKDsZAp7e0XyqFxKgEEYbO1mHPgLOCNwyIb+-IALGCD7AHU37Xh8GkwHB-DZ-toAG7-Zx7MD4NwgCF5ACSUHmsH8YCM2i0ZzR-hgzkqLwGIA0pni308Gj25Lg8SMEKiagiDJgYzwOxJ+TSRgZex2Rm8QnMUEW5IiMG+2iEMDeyAAHAAGNRGODDbRqjm4Lm9RpqGDYZjS2XypAAJjUOXiABUjbY+qS4EFmHN5kJQtgTjk8HAAGIQIw7S4ZPbYW4QEAAXyjQA)

```sh
# Options (if any):
--parser markdown
```

**Input:**

```markdown
---
tags: ["aaaaaaaaaaaaaaaa", "bbbbbbbbbbbbbbb", "cccccccccc", "ddddd", "eeeeeeeeeee"]
---
```

**Output:**

```markdown
---
tags:
  ["aaaaaaaaaaaaaaaa", "bbbbbbbbbbbbbbb", "cccccccccc", "ddddd", "eeeeeeeeeee"]
---
```

**Expected behavior:**

```markdown
---
tags:
  - "aaaaaaaaaaaaaaaa"
  - "bbbbbbbbbbbbbbb"
  - "cccccccccc"
  - "ddddd"
  - "eeeeeeeeeee"
---
```

This is not far off from https://github.com/prettier/prettier/issues/3287.

Basically, `prettier==3.0.0` is incorrectly wrapping long `list`s in Markdown, and this breaks Hugo's YAML frontmatter.

**Comment by sosukesuzuki** [comment]
Thank you for reporting this! I have a question. I think both codes are valid YAML, but does that mean Hugo's Markdown parser can't parse it?

**Comment by jamesbraza** [comment]
Oh wow, actually I didn't realize both were valid YAML. I didn't know one could have an array on the next line like that.

Hugo `v0.116.0` works with both the actual and expected outputs in the original post, so perhaps this isn't actually a `prettier` issue after all!... having multi-line YAML array is a visual preference.

Feel free to close this out 👍

**Comment by fisker** [comment]
I can't reproduce.

**Prettier 3.0.2**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBadAdWBDA5gZyQAIBtDEbSq6m7cgGiPICNW32OGnwxe++u5ACYiRgkHElTpccgF0s6VFhD0QEAA4wAltHzJQ2AE5GIAdwAKxhPpTYANmewBPfWuZHsYANZwYAZWwAWzgAGW0oOGQAMwd8OHdPHz9-DS8I3GQYIwBXBJB4oO0s3Pz8DPs4AEUciHgYuPyAK3wAD38K6tr6pFj7eLUAR264C1MNWwp8VEi4ITnVEGzsbXsMgGEIIKDsZAp7e0XyqFxKgEEYbO1mHPgLOCNwyIb+-IALGCD7AHU37Xh8GkwHB-DZ-toAG7-Zx7MD4NwgCF5ACSUHmsH8YCM2i0ZzR-hgzkqLwGIA0pni308Gj25Lg8SMEKiagiDJgYzwOxJ+TSRgZex2Rm8QnMUEW5IiMG+2iEMDeyAAHAAGNRGODDbRqjm4Lm9RpqGDYZjS2XypAAJjUOXiABUjbY+qS4EFmHN5kJQtgTjk8HAAGIQIw7S4ZPbYW4QEAAXyjQA)

```sh
--parser markdown
```

**Input:**

```markdown
---
tags: ["aaaaaaaaaaaaaaaa", "bbbbbbbbbbbbbbb", "cccccccccc", "ddddd", "eeeeeeeeeee"]
---

```

**Output:**

```markdown
---
tags:
  ["aaaaaaaaaaaaaaaa", "bbbbbbbbbbbbbbb", "cccccccccc", "ddddd", "eeeeeeeeeee"]
---

```

**Comment by jamesbraza** [comment]
@fisker that is a repro 👌 , maybe re-read the original post.  I expected the output to look like this:

```md
---
tags:
  - "aaaaaaaaaaaaaaaa"
  - "bbbbbbbbbbbbbbb"
  - "cccccccccc"
  - "ddddd"
  - "eeeeeeeeeee"
---
```

However, I learned today that it doesn't affect Hugo, it works with either output formatting

**Comment by fisker** [comment]
Sorry, misread. We don't do such change, since their AST are different.  "flowSequence" vs "sequence"
